imports/wasi_snapshot_preview1/fs_test.go

```diff
@@ -1197,7 +1266,32 @@ func Test_fdReaddir(t *testing.T) {
 				}
 			},
 			buf: 0, bufLen: 27, // length is long enough for one more.
-			cookie:          3,  // previous d_next
+			cookie:          2,  // previous d_next
+			expectedBufused: 27, // length to read exactly one more.
+			expectedMem:     dirent3,
+			expectedReadDir: &internalsys.ReadDir{
+				CountRead: 3,
+				Entries:   testDirEntries[2:],
+			},
+		},
+		{
+			name: "read third and beyond",
```

Comment:
I have added this test to validate what happens if it hits the end of the entries with a larger buffer.